### DISCUSSION TITLE
Pdf file

### DISCUSSION THREAD

**Discussion by rafikApp** [Budibase/budibase#7263] [Help] [answered]
Hello everyone.
Is there anything new concerning the creation of a pdf file from a form ?

**Comment by R2bEEaton** [comment]
Not much. You can follow the discussion at #1372.

What do you mean create a PDF from a form? Perhaps I can come up with a hacky way to do this if you described it further.

**Reply by kasgithub8** [reply]
@R2bEEaton 
If you're able to create a pdf from a simple budibase form then go for it. 😄

**Reply by R2bEEaton** [reply]
@kasgithub8 Alright, I went for it and was able to do something fairly good, but pretty limited.

Check #7278 for an explanation and example.

**Reply by rafikApp** [reply]
Hi @R2bEEaton 
I have created a form that will be completed and signed by the client.
After the click on save the data will be stored as well as I want to create a pdf from this data.

**Reply by R2bEEaton** [reply]
> Hi @R2bEEaton 
> I have created a form that will be completed and signed by the client.
> After the click on save the data will be stored as well as I want to create a pdf from this data.

Okay try #7278. If that's not what you're going for then I guess there's nothing you can do but keep waiting for official support.

**Reply by rafikApp** [reply]
Ok, i will try this 
thank you @R2bEEaton

**Reply by rafikApp** [reply]
Hi @R2bEEaton 
Thank you it works.
Do you have a proposal for the signature ? because the form must be signed.
And thank you again.

**Reply by R2bEEaton** [reply]
You could add a text box and ask the user the type their name. Or add a file upload box where they can upload an image which includes a signature. That way it would show up in the PDF. The text box would be easiest and probably look the nicest.

It all depends on what the signature is for and how you want it. If you could elaborate on that I can provide more specific recommendations or ideas.

**Comment by R2bEEaton** [comment, marked as answer]
@rafikApp

I just spent a little while to coming up with a solution using an exploit to achieve something like what you might want. Its main limitation is that text is not selectable on the PDF.

Please refer to #7278 for my explanation and example.
